# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/debug/bug1768660.js
# Commit: d86140d2fd38
# Full Hash: d86140d2fd38138052eba5164efa5e1ec6c3fd0f
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2022-05-18 03:14:37
# Description:
#   Bug 1768660: Skip fewer values in buildExpressionStack r=jandem
#   
#   For the innermost frame of debugger mode bailouts, the current approach effectively uses an allow-list to decide which stack slots need to be recovered. This is fragile to any future bytecode changes that keep values alive on the expression stack. It's also unnecessary: if we just recover slots that are included in the snapshot, and skip slots that don't have allocations, everything works out.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D146525
# ==============================================================================

diff -r 6bca20cf5ff7 -r d86140d2fd38 js/src/jit-test/tests/debug/bug1768660.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/debug/bug1768660.js	Tue May 17 18:17:51 2022 +0000
@@ -0,0 +1,10 @@
+a = newGlobal({newCompartment: true});
+a.parent = this;
+a.eval("Debugger(parent).onExceptionUnwind=function(){}");
+(function() {
+  try {} finally {
+    try {
+      throw 9;
+    } catch {}
+  }
+})()